Lib/datetime.py

```diff
@@ -1414,21 +1414,26 @@ def __repr__(self):
             s = s[:-1] + ", fold=1)"
         return s
 
-    def isoformat(self, timespec='auto'):
+    def isoformat(self, timespec='auto', use_utc_designator=False):
         """Return the time formatted according to ISO.
 
         The full format is 'HH:MM:SS.mmmmmm+zz:zz'. By default, the fractional
-        part is omitted if self.microsecond == 0.
+        part is omitted if self.microsecond == 0. The UTC offset will be
+        replaced with a single "Z" if use_utc_designator is True and
+        self.tzname() is exactly "UTC"
 
         The optional argument timespec specifies the number of additional
         terms of the time to include. Valid options are 'auto', 'hours',
         'minutes', 'seconds', 'milliseconds' and 'microseconds'.
         """
         s = _format_time(self._hour, self._minute, self._second,
                           self._microsecond, timespec)
-        tz = self._tzstr()
-        if tz:
-            s += tz
+        if use_utc_designator and "UTC" == self.tzname():
```

Comment:
Minor thing: this is a strange style for a comparison in Python.  The mistake of `if something = oops_assigned_it` doesn’t exist, so the natural order `if self.tzname() == "UTC"` can be used.

Reply:
That's fair. I'm not gonna polish this further until the datetime maintainers weigh in on whether they want to pursue this approach at all, but if they agree to this approach I'll make this change.